azurelinuxagent/pa/provision/default.py

```diff
@@ -219,13 +219,16 @@ def handle_provision_guest_agent(self, provision_guest_agent):
     def provision(self, ovfenv):
         logger.info("Handle ovf-env.xml.")
         try:
-            logger.info("Set hostname [{0}]".format(ovfenv.hostname))
-            self.osutil.set_hostname(ovfenv.hostname)
 
-            logger.info("Publish hostname [{0}]".format(ovfenv.hostname))
-            self.osutil.publish_hostname(ovfenv.hostname)
+            if conf.get_provisioning_hostname_set:
+                logger.info("Set hostname [{0}]".format(ovfenv.hostname))
+                self.osutil.set_hostname(ovfenv.hostname)
 
-            self.config_user_account(ovfenv)
+                logger.info("Publish hostname [{0}]".format(ovfenv.hostname))
+                self.osutil.publish_hostname(ovfenv.hostname)
+
+            if conf.get_provisioning_user_account_set:
```

Comment:
same here

Reply:
Hmmm, not sure how I missed this. But thanks for catching.. i've updated my PR